File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p02924/p02924_num4_parsed.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main() {
    // Read the input
    int N;
    cin >> N;

    // Create a vector of N numbers
    vector<int> numbers(N);
    for (int i = 0; i < N; i++) {
        numbers[i] = i + 1;
    }

    // Sort the vector in descending order
    sort(numbers.begin(), numbers.end(), greater<int>());

    // Calculate the maximum possible value
    long long sum = 0;
    for (int i = 0; i < N; i++) {
        sum += i % numbers[i];
    }

    // Print the result
    cout << sum << endl;

    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p02924/p02924_num4_parsed.py
import math

def calculate_max_sum(N):
    # Generate the permutation of {1, 2, ..., N}
    permutation = list(range(1, N+1))
    
    # Calculate the maximum possible value of M_1 + M_2 + ... + M_N
    max_sum = 0
    for i in range(1, N+1):
        remainder = i % permutation[i-1]
        max_sum += remainder
    
    return max_sum

# Read the input N
N = int(input())

# Calculate and print the maximum possible value of M_1 + M_2 + ... + M_N
print(calculate_max_sum(N))